File: bank-frontend/server/controllers/accountController.js
const { spring } = require("../services/bankService");

const addAccount = async (req, res, next) => {
  try {
    const { data } = await spring.post("/accounts", req.body);
    res.json(data);
  } catch (e) {
    next(e);
  }
};

const getAllAccounts = async (req, res, next) => {
  try {
    const { data } = await spring.get("/accounts");
    res.json(data);
  } catch (e) {
    next(e);
  }
};

const getAccountById = async (req, res, next) => {
  try {
    const { accountId } = req.params;
    const { data } = await spring.get(`/accounts/${accountId}`);
    res.json(data);
  } catch (e) {
    next(e);
  }
};

const updateAccount = async (req, res, next) => {
  try {
    const { accountId } = req.params;
    const { data } = await spring.put(`/accounts/${accountId}`, req.body);
    res.json(data);
  } catch (e) {
    next(e);
  }
};

const deleteAccount = async (req, res, next) => {
  try {
    const { accountId } = req.params;
    await spring.delete(`/accounts/${accountId}`);
    res.json({ message: "Deleted successfully" });
  } catch (e) {
    next(e);
  }
};

const getAccountsByMember = async (req, res, next) => {
  try {
    const { memberId } = req.params;
    const { data } = await spring.get(`/accounts/member/${memberId}`);
    res.json(data);
  } catch (e) {
    next(e);
  }
};

module.exports = {
  addAccount,
  getAllAccounts,
  getAccountById,
  updateAccount,
  deleteAccount,
  getAccountsByMember,
};

File: quarkus/backend-api/src/main/java/org/digio/bank/controller/AccountController.java
package org.digio.bank.controller;

import jakarta.inject.Inject;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

import org.digio.bank.dto.OpenAccount;
import org.digio.bank.service.AccountService;
import org.digio.entity.model.Account;

import java.util.List;

@Path("/accounts")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class AccountController {

    private final AccountService accountService;

    @Inject
    public AccountController(AccountService accountService) {
        this.accountService = accountService;
    }

    @GET
    public List<Account> getAllAccounts() {
        return accountService.getAllAccounts();
    }

    @GET
    @Path("/{accountId}")
    public Response getAccountById(@PathParam("accountId") Long accountId) {
        Account account = accountService.getAccountById(accountId);
        return account != null ? Response.ok(account).build() : Response.status(Response.Status.NOT_FOUND).build();
    }

    @GET
    @Path("/member/{memberId}")
    public List<Account> getAccountsByMember(@PathParam("memberId") Long memberId) {
        return accountService.getAccountsByMember(memberId);
    }

    @POST
    public Response addAccount(@HeaderParam("x-member-id") Long loginMemberId, OpenAccount request) {
        if (loginMemberId == null) {
            return Response.status(Response.Status.UNAUTHORIZED).build();
        }

        if (request.getMemberId() == null) {
            throw new IllegalArgumentException("กรุณากรอกเลขบัตรประชาชน");
        }

        if (!loginMemberId.equals(request.getMemberId())) {
            throw new IllegalArgumentException("เลขบัตรประชาชนไม่ตรงกับผู้ใช้งานที่ล็อกอิน");
        }

        Account created = accountService.createAccount(request);
        return Response.status(Response.Status.CREATED).entity(created).build();
    }

    @PUT
    @Path("/{accountId}")
    public Response updateAccount(@PathParam("accountId") Long accountId, Account update) {
        Account updated = accountService.updateAccount(accountId, update);
        return updated != null ? Response.ok(updated).build() : Response.status(Response.Status.NOT_FOUND).build();
    }

    @DELETE
    @Path("/{accountId}")
    public Response deleteAccount(@PathParam("accountId") String accountId) {
        boolean deleted = accountService.deleteAccount(accountId);
        return deleted ? Response.noContent().build() : Response.status(Response.Status.NOT_FOUND).build();
    }
}
